File: 스택,큐.힙/다리를지나는트럭(PRO).py
from collections import deque

def solution(bridge_length, weight, truck_weights):
    q = deque([0 for _ in range(bridge_length-1)])
    
    t = 1
    i = 1
    sum = truck_weights[0]
    
    q.append(truck_weights[0])
    
    while sum > 0:
        sum -= q.popleft()
        
        
        if i < len(truck_weights) and sum + truck_weights[i] <= weight:
            q.append(truck_weights[i])
            i += 1
            sum += truck_weights[i]
        else:
            q.append(0)
        
        t += 1
            
    return t

File: 스택,큐.힙/다리를지나는트럭(PRO).kt
import java.util.*

class Solution {
    fun solution(bridge_length: Int, weight: Int, truck_weights: IntArray): Int {
        var q: Queue<Int> = LinkedList()
        
        for (i in 0 until bridge_length - 1) {
            q.offer(0)
        }
        q.offer(truck_weights[0])
        
        var t = 1
        var i = 1
        
        while (q.sum() > 0) {
            q.poll()
            
            if (i < truck_weights.size && q.sum() + truck_weights[i] <= weight) {
                q.offer(truck_weights[i])
                i += 1
            } else {
                q.offer(0)
            }
            t += 1
        }
        
        return t
    }
}